### PR TITLE
AITRIAGE-4849: added a signature for missing OSTree pivot

### PR DESCRIPTION
**Pull request by danielerez**
Added a signature that seaches pivot URL in rpm-ostree/status for each controle-plane host.
If the pivot is missing, it could be related to this issue opened on MCO: https://issues.redhat.com/browse/MCO-449

### CODE REVIEW

File: tools/add_triage_signature.py [hunk 1]
```diff
@@ -2327,6 +2327,66 @@ def _process_ticket(self, url, issue_key):
...
+                    OrderedDict(
+                        Hostname=hostname,
+                        IP=node_ip,
+                    )
+                )
+
+        if len(hosts) != 0:
+            report = dedent(
+                """
+            MCO didn't invoke OSTree pivot on the following bootstrap control-plane hosts:
```
[the remark this answers is not in the document]
Comment (omertuc) [reply tools/add_triage_signature.py:2382]: (Those 2 control plane nodes are not bootstrap)

File: tools/add_triage_signature.py [hunk 2]
```diff
@@ -2327,6 +2327,66 @@ def _process_ticket(self, url, issue_key):
...
+            [Could be related to this issue in MCO: https://issues.redhat.com/browse/MCO-449]
+            """
+            )
+            report += self._generate_table_for_report(hosts)
+            self._update_triaging_ticket(report)
```
Comment (omertuc) [inline tools/add_triage_signature.py:2387]: nit: `report` variable seems unnecessary 

```suggestion
            self._update_triaging_ticket(dedent(
                f"""
            MCO didn't invoke OSTree pivot on the following bootstrap control-plane hosts:
            [Could be related to this issue in MCO: https://issues.redhat.com/browse/MCO-449]
            {self._generate_table_for_report(hosts)}
            """
            ))
```

File: tools/add_triage_signature.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielerez) [reply tools/add_triage_signature.py:2382]: Right, forgot the 'non':/ Removed.

File: tools/add_triage_signature.py (hunk 2 above)
Comment (danielerez) [reply tools/add_triage_signature.py:2387]: done

File: tools/add_triage_signature.py [hunk 3]
```diff
@@ -2327,6 +2327,67 @@ def _process_ticket(self, url, issue_key):
...
+                        IP=node_ip,
+                    )
+                )
+
+        if len(hosts) != 0:
+            self._update_triaging_ticket(
+                dedent(
+                    f"""
+            MCO didn't invoke OSTree pivot on the following non-bootstrap control-plane hosts:
+            [Could be related to this issue in MCO: https://issues.redhat.com/browse/MCO-449]
```
Comment (romfreiman) [inline tools/add_triage_signature.py:2384]: please update to the ocpbug ticket

File: tools/add_triage_signature.py [hunk 4]
```diff
@@ -2327,6 +2327,67 @@ def _process_ticket(self, url, issue_key):
...
+
+    PIVOT_URL_PATTERN = re.compile(r"pivot://.*")
+    CONTROL_PLANE_PATH = "*_bootstrap_*.tar.gz/logs_host_*/log-bundle-*.tar.gz/log-bundle-*/control-plane"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            function_impact_label="missing_pivot_ostree",
+            comment_identifying_string="h1. MCO didn't pivot some hosts",
```
Comment (romfreiman) [inline tools/add_triage_signature.py:2343]: do we want to negate on @javipolo signature?
https://github.com/openshift-assisted/assisted-installer-deployment/pull/328/files
although it's not so common, so not sure it worth the effort.

File: tools/add_triage_signature.py [hunk 5]
```diff
@@ -2327,6 +2327,67 @@ def _process_ticket(self, url, issue_key):
...
+    """
+    This signature looks for missing pivot URL in rpm-ostree status of the control-plane hosts
+    """
+
+    PIVOT_URL_PATTERN = re.compile(r"pivot://.*")
+    CONTROL_PLANE_PATH = "*_bootstrap_*.tar.gz/logs_host_*/log-bundle-*.tar.gz/log-bundle-*/control-plane"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
```
Comment (romfreiman) [inline tools/add_triage_signature.py:2340]: Just a though: are we running it only on masters or also on workers? 
control plane is only masters, right?

what will happen if a worker will suffer from similar issue? Lets not block this PR, but worth thinking about it

**Comment by romfreiman** [comment]
did u run it on 30 days of tickets? How many such cases u identified?

**Comment by danielerez** [comment]
> did u run it on 30 days of tickets? How many such cases u identified?

Yeah, 15 tickets were affected out of 249 ticket in the last 30 days (not sure if all indeed had the same exact issue)

File: tools/add_triage_signature.py (hunk 5 above)
Comment (omertuc) [reply tools/add_triage_signature.py:2340]: Bootstrap doesn't gather logs about workers, so we have no information about them anyway

Comment (omertuc) [reply tools/add_triage_signature.py:2340]: We *might* have this information in a must-gather, but I'm not really sure of that

Comment (eranco74) [reply tools/add_triage_signature.py:2340]: I don't think we can get this info for workers.
Unless we update installer-gather code or scp files from the workers ourserlves.

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**